### ISSUE TITLE
Please Help: Sony Hx300

### ISSUE THREAD

**Issue by amitpandey2194** [ma1co/Sony-PMCA-RE#198]
Querying MTP device
Sony Corporation DSC-HX300 is a camera in MTP mode

Model: DSC-HX300
Product code: 0080813110
Serial number: 04520210
Firmware version: 1.00

Using drivers Windows-MSC, Windows-MTP
Looking for Sony devices

Querying MTP device
Sony Corporation DSC-HX300 is a camera in MTP mode

Switching to app install mode
Traceback (most recent call last):
File "C:\projects\sony-pmca-re\pmca-gui.py", line 76, in do
File "C:\projects\sony-pmca-re\pmca\commands\usb.py", line 274, in installCommand
File "C:\projects\sony-pmca-re\pmca\commands\usb.py", line 37, in switchToAppInstaller
File "C:\projects\sony-pmca-re\pmca\usb\sony.py", line 306, in switchToAppInstaller
File "C:\projects\sony-pmca-re\pmca\usb\sony.py", line 244, in sendCommand
File "C:\projects\sony-pmca-re\pmca\usb\sony.py", line 98, in sendSonyExtCommand
File "C:\projects\sony-pmca-re\pmca\usb_init.py", line 81, in _checkResponse
pmca.usb.InvalidCommandException: MTP error 0x2006

I also tried MTP as well as Mass storage option on the camera

**Comment by ma1co** [comment]
As a first step, we have to dump the firmware. We'll do this using the firmware update mode. First, we have brute force the model id of your camera. This won't install anything on your device. If you're interested to help, here's what you can do:

* If you're on windows, download the newest development build of pmca-console [here](https://ci.appveyor.com/project/ma1co/sony-pmca-re/build/artifacts) or pull the most recent commit in the repository.
* Download [0x31020000.zip](https://github.com/ma1co/Sony-PMCA-RE/files/5048140/0x31020000.zip) and place it in the same folder (no need to unpack). It contains a bunch of firmware images which might match your camera.
* Power on your camera and connect it over USB to your computer.
* Open a command prompt and run the following command: `pmca-console guess_firmware -f 0x31020000.zip`
* With a bit of luck, the script should find the model id of your camera.

Let me know if you need any help.

**Comment by amitpandey2194** [comment]
@ma1co thanks a lot for responding.
I did what you said: Here is the output when I ran the above command.
Using drivers Windows-MSC, Windows-MTP
Looking for Sony devices

Querying MTP device
Sony Corporation DSC-HX300 is a camera in MTP mode

Getting device info
Model name: DSC-HX300

Trying 100 firmware images
Firmware update error: Invalid version
Success: Found matching file: gen2_0x31020050_0.dat
PS C:\Users\amitp\Desktop\SonyCamera> .\pmca-console-v0.17-30-gb1faddd-win.exe guess_firmware -f 0x31020000.zip
Using drivers Windows-MSC, Windows-MTP
Looking for Sony devices

Querying MTP device
Sony Corporation DSC-HX300 is a camera in MTP mode

Getting device info
Model name: DSC-HX300

Trying 100 firmware images
Firmware update error: Invalid version
Success: Found matching file: gen2_0x31020050_0.dat

What should I do next?

**Comment by ma1co** [comment]
All good, there is only one boot partition on these older cameras. Thank you for the help!
The camera is now supported in pmca-console / pmca-gui.

**Comment by amitpandey2194** [comment]
Hi @ma1co,
I tried installing the OpenTweak Memories Tweak,
It is still giving me the same error as above. Do I need a specific version of the pmca where you updated the support. Do I need to install the firmware? I am not so clear about the next steps. I just went ahead and downloaded the build from [here](https://ci.appveyor.com/project/ma1co/sony-pmca-re/build/artifacts) and tried installing but it still gives the error.

Using drivers Windows-MSC, Windows-MTP
Looking for Sony devices

Querying MTP device
Sony Corporation DSC-HX300 is a camera in MTP mode

Switching to app install mode
Traceback (most recent call last):
  File "pmca-gui.py", line 76, in do
  File "pmca\commands\usb.py", line 277, in installCommand
  File "pmca\commands\usb.py", line 37, in switchToAppInstaller
  File "pmca\usb\sony.py", line 306, in switchToAppInstaller
  File "pmca\usb\sony.py", line 244, in _sendCommand
  File "pmca\usb\sony.py", line 98, in sendSonyExtCommand
  File "pmca\usb\__init__.py", line 81, in _checkResponse
pmca.usb.InvalidCommandException: MTP error 0x2006
